### PR TITLE
✨ sequin.yml

### PR DESCRIPTION
**Pull request by RTLS**

### CODE REVIEW

**Comment by RTLS** [comment]
* **#485** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/485?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/config.ex [hunk 1]
```diff
@@ -0,0 +1,258 @@
+defmodule Sequin.Config do
```
Comment (acco) [inline lib/sequin/config.ex:1]: Replace Nouns w/ Verbs

YAMLLoader

File: lib/sequin/config.ex [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  alias Sequin.Databases.PostgresDatabase
+  alias Sequin.Databases.Sequence
+  alias Sequin.Error.NotFoundError
+  alias Sequin.Repo
+
+  require Logger
+
+  @app :sequin
+
+  def apply_config do
```
Comment (acco) [inline lib/sequin/config.ex:14]: check self_hosted? // or config_allowed?

File: lib/sequin/config.ex [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  defp do_find_or_create_account!(%{}) do
+    case Accounts.find_account(name: "Configured by Sequin") do
+      {:ok, account} ->
+        account
+
+      {:error, %NotFoundError{}} ->
+        {:ok, account} = Accounts.create_account(@account_defaults)
+        account
+    end
+  end
```
Comment (acco) [inline lib/sequin/config.ex:86]: turn into a helpful error message if missing account

File: lib/sequin/config.ex [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  ###############
+
+  @database_defaults %{
+    "username" => "postgres",
+    "password" => "postgres",
+    "port" => 5432
+  }
+
+  defp find_and_update_or_create_databases!(account_id, %{"databases" => databases}) do
+    Logger.info("Creating databases: #{inspect(databases, pretty: true)}")
```
Comment (acco) [inline lib/sequin/config.ex:99]:
```suggestion
    Logger.info("Upserting databases: #{inspect(databases, pretty: true)}")
```

File: lib/sequin/config.ex [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  ## Databases ##
+  ###############
+
+  @database_defaults %{
+    "username" => "postgres",
+    "password" => "postgres",
+    "port" => 5432
+  }
+
+  defp find_and_update_or_create_databases!(account_id, %{"databases" => databases}) do
```
Comment (acco) [inline lib/sequin/config.ex:98]:
```suggestion
  defp upserting_databases!(account_id, %{"databases" => databases}) do
```

File: lib/sequin/config.ex [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    |> case do
+      {:ok, sequence} -> sequence
+      {:error, error} when is_exception(error) -> raise "Failed to create sequence: #{Exception.message(error)}"
+      {:error, %Ecto.Changeset{} = changeset} -> raise "Failed to create sequence: #{inspect(changeset)}"
+    end
+  end
+
+  defp update_sequence!(%Sequence{} = sequence, %PostgresDatabase{} = database) do
+    Databases.update_sequence_from_db(sequence, database)
+  end
```
Comment (acco) [inline lib/sequin/config.ex:191]: ?

File: lib/sequin/databases/databases.ex [hunk 7]
```diff
@@ -141,10 +141,13 @@ defmodule Sequin.Databases do
 
   # Sequences
 
-  def get_sequence_for_account(account_id, sequence_id) do
-    account_id
-    |> Sequence.where_account()
-    |> Sequence.where_id(sequence_id)
+  def get_sequence_for_account(account_id, params \\ []) do
```
Comment (acco) [inline lib/sequin/databases/databases.ex:144]:
```suggestion
  def find_sequence_for_account(account_id, params \\ []) do
```

**Pull request merged**